src/agora/txrx/txrx_DPDK.cc

```diff
@@ -64,26 +67,39 @@ PacketTXRX::PacketTXRX(Config* cfg, size_t core_offset,
   tx_ptoks_ = tx_ptoks;
 }
 
-PacketTXRX::~PacketTXRX() { rte_mempool_free(mbuf_pool); }
+PacketTXRX::~PacketTXRX() { rte_mempool_free(mbuf_pool_); }
 
-bool PacketTXRX::StartTxRx(Table<char>& buffer, Table<int>& buffer_status,
-                           size_t packet_num_in_buffer,
+bool PacketTXRX::StartTxRx(Table<char>& buffer, size_t packet_num_in_buffer,
                            Table<size_t>& frame_start, char* tx_buffer,
                            Table<complex_float>& calib_dl_buffer,
                            Table<complex_float>& calib_ul_buffer) {
   unused(calib_dl_buffer);
   unused(calib_ul_buffer);
-  buffer_ = &buffer;
-  buffer_status_ = &buffer_status;
-  frame_start_ = &frame_start;
 
-  packet_num_in_buffer_ = packet_num_in_buffer;
+  frame_start_ = &frame_start;
+  buffers_per_socket_ = packet_num_in_buffer / socket_thread_num_;
   tx_buffer_ = tx_buffer;
 
+  rx_packets_.resize(socket_thread_num_);
+  for (size_t i = 0; i < socket_thread_num_; i++) {
+    rx_packets_.at(i).reserve(buffers_per_socket_);
+#if defined(USE_DPDK_MEMORY)
+    unused(buffer);
+    rx_packets_.at(i).resize(buffers_per_socket_);
+#else
+    for (size_t number_packets = 0; number_packets < buffers_per_socket_;
+         number_packets++) {
+      auto* pkt_loc = reinterpret_cast<Packet*>(
+          buffer[i] + (number_packets * cfg_->PacketLength()));
+      rx_packets_.at(i).emplace_back(pkt_loc);
+    }
+#endif  // defined(USE_DPDK_MEMORY)
+  }
+
   unsigned int lcore_id;
   size_t worker_id = 0;
   // Launch specific task to cores
-  RTE_LCORE_FOREACH_SLAVE(lcore_id) {
+  RTE_LCORE_FOREACH_WORKER(lcore_id) {
```

Comment:
This function is not defined in DPDK version 20.02.1. Shall we add a MACRO for handling this?

Reply:
I think we will just revert it to the correct call for 20.02 since in 20.11 it just produces a warning.  We can fix it when official support for 20.11 is added